packages/office-ui-fabric-react/src/utilities/decorators/withResponsiveMode.tsx

```diff
@@ -59,48 +59,42 @@ export function withResponsiveMode<TProps extends { responsiveMode?: ResponsiveM
 ): any {
   const resultClass = class WithResponsiveMode extends BaseDecorator<TProps, IWithResponsiveModeState> {
     private _events: EventGroup;
+    private _async: Async;
 
     constructor(props: TProps) {
       super(props);
       this._events = new EventGroup(this);
+      this._async = new Async(this);
       this._updateComposedComponentRef = this._updateComposedComponentRef.bind(this);
 
-      this.state = {
-        responsiveMode: getInitialResponsiveMode(),
-      };
+      // prevent excessive rerenders by using debounce
+      this._onResize = this._async.debounce(this._onResize, 500);
     }
 
     public componentDidMount(): void {
       this._events.on(window, 'resize', this._onResize);
-      this._onResize();
     }
 
     public componentWillUnmount(): void {
       this._events.dispose();
     }
 
     public render(): JSX.Element | null {
-      const { responsiveMode } = this.state;
-
-      return responsiveMode === ResponsiveMode.unknown ? null : (
-        <ComposedComponent
-          ref={this._updateComposedComponentRef}
-          responsiveMode={responsiveMode}
-          {...(this.props as any)}
-        />
+      return (
+        <WindowContext.Consumer>
+          {({ window }) => (
+            <ComposedComponent
+              ref={this._updateComposedComponentRef}
+              responsiveMode={getResponsiveMode(window)}
```

Comment:
I like this in general, but I wonder if calculating the responsive mode on every render could somehow cause perf problems? Mainly a concern if `window.innerWidth` is one of the things that's expensive to access. (I don't remember the list of problematic properties offhand and I'm not sure what to call it in a search...)

Reply:
`window.innerWidth` access may cause forced browser reflow https://gist.github.com/paulirish/5d52fb081b3570c81e3a. Not sure if that's what you are referring to above.

Reply:
Yes it is, thanks! So we should probably find a way to ensure the calculation only runs on resize.

Reply:
As it turns out you can access class contexts via `static context` I've moved to that and it has cleared all the problems.